### PR TITLE
check k8s and ocp version for telemetry on deploy and on activation send kn and function version

### PR DESCRIPTION
**Pull request by sudhirverma**

### CODE REVIEW

File: src/check-cluster.ts [hunk 1]
```diff
@@ -6,11 +7,19 @@
...
 import { KubectlAPI } from './cli/kubectl-api';
+import { telemetryLog } from './telemetry';
 
-export async function checkOpenShiftCluster(): Promise<boolean> {
+export async function checkOpenShiftCluster(checkClusterVersion?: boolean): Promise<boolean> {
   try {
     const result = await knExecutor.execute(KubectlAPI.checkOcpCluster(), process.cwd(), false);
     if (result?.stdout?.trim()) {
+      if (checkClusterVersion) {
+        telemetryLog(
```
Comment (lstocchi) [inline src/check-cluster.ts:17]: Why sending ocp version only if user performs a deploy? Wouldn't it be better to send it during activation or cluster switch? `The user is connected to kubernetes or ocp version blabla`? Then you know all following actions are performed to it. If the user executes 2-3 deploys the ocp version would always be the same. Why sending it each time?

File: src/cli/cli-config.ts [hunk 2]
```diff
@@ -129,6 +129,10 @@ export class CmdCliConfig {
     return reqs;
   }
 
+  static getCliLocation(cmd: string): string {
```
Comment (lstocchi) [inline src/cli/cli-config.ts:132]: I don't find any usage of this

File: src/cli/func-api.ts [hunk 3]
```diff
@@ -84,8 +89,17 @@ export class FuncAPI {
 
   static async deployFunc(location: string, image: string, namespace: string): Promise<CliCommand> {
     const deployCommand = ['deploy', '-p', `${quote}${location}${quote}`, '-i', image, '-v', '-n', namespace];
-    if (await checkOpenShiftCluster()) {
+    if (await checkOpenShiftCluster(true)) {
       deployCommand.push('-r ""');
+    } else {
+      const result = await knExecutor.execute(KubectlAPI.printVersion(), process.cwd(), false);
+      if (result?.stdout?.trim()) {
+        telemetryLog(
```
Comment (lstocchi) [inline src/cli/func-api.ts:97]: Same comment as above

File: src/extension.ts [hunk 4]
```diff
@@ -126,6 +139,18 @@ export async function activate(extensionContext: vscode.ExtensionContext): Promi
   disposable.forEach((value) => extensionContext.subscriptions.push(value));
 }
 
+async function sendVersionToTelemetry(commandId: string, cmd: string): Promise<void> {
+  const result = await executeCmdCli.execute(createCliCommand(cmd, 'version'));
+  if (result.error) {
+    telemetryLogError(commandId, result.error);
+  }
```
Comment (lstocchi) [inline src/extension.ts:146]: Shouldn't you return/exit after the `telemetryLogError`? If it happens i guess there is no version to report.

**Review by lstocchi — CHANGES_REQUESTED** [review]

File: src/extension.ts (hunk 4 above)
Comment (sudhirverma) [reply src/extension.ts:177]: this is telemetry specific so if there is an error we can see it in the telemetry log.

File: src/check-cluster.ts (hunk 1 above)
Comment (sudhirverma) [reply src/check-cluster.ts]: done

File: src/cli/cli-config.ts (hunk 2 above)
Comment (sudhirverma) [reply src/cli/cli-config.ts]: Removed

File: src/cli/func-api.ts (hunk 3 above)
Comment (sudhirverma) [reply src/cli/func-api.ts]: done

File: src/extension.ts (hunk 4 above)
Comment (lstocchi) [reply src/extension.ts:177]: yeah i was just thinking that if result.error has content, the result.stdout will likely not have the version in it so the following `telemetryLog` would make no sense.

**Review by lstocchi — APPROVED** [review]
LGTM

File: src/extension.ts (hunk 4 above)
Comment (sudhirverma) [reply src/extension.ts:177]: thanks I have fixed :)

**Pull request merged**